File: Data Structures & Algorithms/counting-bits/submission-2.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        if n == 2:
            return [0, 1, 1]
        res = [0] * (n + 1)
        
        nearest = 2
        for i in range(1, n + 1):
            if i == nearest * 2:
                nearest *= 2
            res[i] = res[i - nearest] + 1

        return res

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> cnts;
        vector<vector<int>> freqs (nums.size() + 1);
        for (int num: nums) {
            cnts[num]++;
        }
        for (const auto& [num, cnt]: cnts) {
            freqs[cnt].push_back(num);
        }
        vector<int> res;
        for (int i = freqs.size() - 1; i >= 0; i--) {
            for (int num : freqs[i]) {
                res.push_back(num);
                if (res.size() == k) {
                    return res;
                }
            }
        }
    }
};
